dipy/reconst/sfm.py

```diff
@@ -225,36 +316,36 @@ def fit(self, data, mask=None):
         # Fitting is done on the relative signal (S/S0):
         flat_S0 = np.mean(flat_data[..., self.gtab.b0s_mask], -1)
         flat_S = flat_data[..., ~self.gtab.b0s_mask] / flat_S0[..., None]
-        flat_mean = np.mean(flat_S, -1)
+        flat_isotropic = self.isotropic(self.gtab).fit(flat_data).predict()
         flat_params = np.zeros((flat_data.shape[0],
                                 self.design_matrix.shape[-1]))
 
         for vox, vox_data in enumerate(flat_S):
             if np.any(np.isnan(vox_data)):
-                flat_params[vox] = (np.zeros(self.design_matrix.shape[-1]))
+                pass
```

Comment:
I've tried to simplify and clarify by adding a comment and a test (https://github.com/arokem/dipy/commit/ac011dfdfbdc5fb197a9131823d7394ed390848e). This is for cases in which the non diffusion-weighted signal is 0, like in voxels outside the brain. In these cases, `ElasticNet` would throw an error, so we pre-empt that by just leaving the parameters as all-zeros.